src/android/SplashScreen.java

```diff
@@ -151,13 +193,40 @@ public Object onMessage(String id, Object data) {
         }
         return null;
     }
+    
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        // Reload splash drawable when orientation changes if so configured.
+        if (newConfig.orientation != orientation) {
+            orientation = newConfig.orientation;
+            reloadDrawable();
+        }
+    }
 
-    private void removeSplashScreen() {
+    private void reloadDrawable() {
+        if (isReloadOnOrientationChange() && splashImageView != null) {
```

Comment:
I think a performance hit would be acceptable here given the ease of configuration. How often are you going to rotate your device while the splash screen is up? Probably close to never, or at most once.